ssz/sedes/list.py

```diff
@@ -0,0 +1,102 @@
+from collections.abc import (
+    Iterable,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes import (
+    address,
+    boolean,
+    hash32,
+    uint32,
+)
+
+
+class List:
+    """
+    A sedes for lists.
+
+    WARNING: Avoid sets if possible, may not always lead to expected results
+    (This is because iteration in sets doesn't always happen in the same order)
+    """
+    LENGTH_BYTES = 4
+
+    def __init__(self, element_sedes):
```

Comment:
If I understand correctly, the empty list sedes just uses `None` as `element_sedes` and trying to serialize/deserialize something non-empty would raise an `AttributeError` which is misleading.

I see two better options, maybe there are more:

1) Add an `EmptyList` sedes that is not related to `List` at all and only works for empty lists (raising `Serialization`/`DeserializationError`s for everything else).
2) Add an explicit argument `empty` to `List.__init__` and check that during serialization/deserialization. Also raise `ValueError`s if one tries to initialize `List` with `empty` and an `element_sedes` or without `empty` and without an `element_sedes`.